core/frontend/src/tools/AccuDrawViewportUI.ts

```diff
@@ -0,0 +1,733 @@
+    ev.stopPropagation();
+
+    if (!isDown)
+      return;
+
+    return this.acceptPartialInput(item, forward);
+  }
+
+  /** @internal */
+  protected async doAcceptInput(ev: KeyboardEvent, isDown: boolean, item: ItemField): Promise<void> {
+    ev.preventDefault();
+    if (!isDown || this.isDynamicKeyinStatus(item))
+      return;
+
+    ev.stopPropagation();
+    return this.acceptPartialInput(item);
+  }
+
+  /** @internal */
+  protected doNewInput(_ev: KeyboardEvent, isDown: boolean, item: ItemField): void {
+    if (!isDown)
+      return;
+
+    this.setPartialKeyinStatus(item, false);
+  }
+
+  /** @internal */
+  protected async doDeleteInput(_ev: KeyboardEvent, isDown: boolean, item: ItemField): Promise<void> {
+    if (isDown)
+      return this.setPartialKeyinStatus(item, false);
+
+    return this.processPartialInput(item);
+  }
+
+  /** @internal */
+  protected processReplacementKey(ev: KeyboardEvent, isDown: boolean, item: ItemField, replacement: string, allowStart: boolean): boolean {
+    if (undefined === this.controls)
+      return false;
+
+    ev.preventDefault()
+    if (!isDown || 0 === replacement.length || this.isDynamicKeyinStatus(item))
+      return true;
+
+    const itemField = this.controls.itemFields[item];
+    if (null === itemField.selectionStart || null === itemField.selectionEnd)
+      return true;
+
+    const currentValue = itemField.value;
+    if (!allowStart && (!currentValue.length || !itemField.selectionStart || !itemField.selectionEnd))
+      return true;
+
+    const selectionStart = (itemField.selectionStart > itemField.selectionEnd ? itemField.selectionEnd : itemField.selectionStart);
+    itemField.value = currentValue.substring(0, selectionStart) + replacement + currentValue.substring(itemField.selectionEnd);
+    itemField.selectionStart = itemField.selectionEnd = selectionStart + 1;
+
+    return false;
+  }
+
+  /** @internal */
+  protected processRepeatedKey(ev: KeyboardEvent, isDown: boolean, item: ItemField, replacement: string): boolean {
+    if (undefined === this.controls)
+      return false;
+
+    const itemField = this.controls.itemFields[item];
+    const currentValue = itemField.value;
+    if (!currentValue.length || !itemField.selectionStart || !itemField.selectionEnd)
+      return false;
+
+    const selectionStart = (itemField.selectionStart > itemField.selectionEnd ? itemField.selectionEnd : itemField.selectionStart);
+    if (selectionStart !== itemField.selectionEnd)
+      return false;
+
+    const selectionPrevious = selectionStart - 1;
+    if (currentValue[selectionPrevious] !== (isDown ? ev.key : replacement))
+      return false;
+
+    ev.preventDefault()
+    if (!isDown)
+      return false;
+
+    itemField.value = currentValue.substring(0, selectionPrevious) + replacement + currentValue.substring(selectionStart);
+    itemField.selectionStart = itemField.selectionEnd = selectionPrevious + 1;
+
+    return false;
+  }
+
+  /** @internal */
```

Comment:
`doProcessOverrideKey` and `doProcessKey` are `@internal` but also `protected`, suggesting subclasses can be expected to override them. If they shouldn't be overridden, make them `private`. If they can be overridden by subclasses outside of itwinjs-core, make them `@public`. If they should only be overridden by subclasses inside itwinjs-core, rename them to use a `Symbol`.

Reply:
Same goes for all the other `@internal` `protected` members like `focusItem`, `doNavigate`, etc.

Reply:
Done.